js/base/functions/generic.js

```diff
@@ -77,14 +77,28 @@ module.exports =
           value2: { someKey: 'value2', anotherKey: 'anotherValue2' },
           value3: { someKey: 'value3', anotherKey: 'anotherValue3' },
       }
+
+       Array objects itself can be array too:
+       array = [
+          [7900.0, 10.0],
+          [7901.0, 8.8],
+       ]
+       key = 0
+
+       Returns:
+      {
+          7900.0: [7900.0, 10.0],
+          7901.0: [7901.0, 8.8],
+      }
     */
 
     , indexBy (x, k, out = {}) {
-
-        for (const v of values (x))
-            if (k in v)
+        const isIntKey = isInteger (k)
```

Comment:
Actually, we can revert this, since:

```
mbp:ccxt igorkroitor$ node
Welcome to Node.js v12.2.0.
Type ".help" for more information.
> const v = [ 'a', '1', 3, '5', 7, undefined, 13 ]
undefined
> let k = 1
undefined
> k in v
true
> k = 0
0
> k in v
true
> k = 10
10
> k in v
false
>
```